src/v/kafka/server/connection_context.cc

```diff
@@ -294,6 +294,8 @@ connection_context::dispatch_method_once(request_header hdr, size_t size) {
               const auto correlation = rctx.header().correlation;
               const sequence_id seq = _seq_idx;
               _seq_idx = _seq_idx + sequence_id(1);
+              auto record_time_quota = _proto.quota_mgr().time_quota_recorder(
```

Comment:
Should the time quota take into account any throttle that was applied? Technically its part of total request time, however not a part of actual work the broker had been doing

Reply:
mm, i don't think so. a good way to determine if it should is to consider if it was a large delay, how would it affect the math?

Reply:
I agree. To answer your question i guess it would skew the results until the max throttle time is hit, holding it there across requests.